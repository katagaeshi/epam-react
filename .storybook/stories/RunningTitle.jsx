import React from 'react';
import { storiesOf } from '@storybook/react';

import {
  configure,
} from '@storybook/react';

import RunningTitle from '../../public/components/ui/RunningTitle';

storiesOf('RunningTitle', module)
  .add('with text', () => (
    <RunningTitle
      text='Example of title'
    />
  ));
