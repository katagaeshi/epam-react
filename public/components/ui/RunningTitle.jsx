// @flow

/*
  As header and footer are the same currently in our program
  it means logically to make one component for both.
*/

import React from 'react';
import injectSheet from 'react-jss';

const styles = {
  runningTitle: {
    'padding-top': '20px',
    'padding-bottom': '20px',
    'background-color': 'gray',
    color: 'pink',
    display: 'block',
  },
};

const defaultText = 'netflixroulette';

type Props = {
  text: string,
  classes: {
    runningTitle: string,
  }
};

const RunningTitle = (props: Props) => <span className={props.classes.runningTitle}>{props.text}</span>;

RunningTitle.defaultProps = {
  text: defaultText,
};

export default injectSheet(styles)(RunningTitle);
