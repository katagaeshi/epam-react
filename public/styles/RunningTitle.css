.RunningTitle {
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: gray;
  color: pink;
  display: block;
}
