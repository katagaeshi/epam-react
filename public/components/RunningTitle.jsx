/*
  As header and footer are the same currently in our program
  it means logically to make one component for both.
*/

import React from 'react';
import PropTypes from 'prop-types';
import '../styles/RunningTitle.css';

const defaultText = 'netflixroulette';

const RunningTitle = props => <span className="RunningTitle">{props.text}</span>;

RunningTitle.defaultProps = {
  text: defaultText,
};

RunningTitle.propTypes = {
  text: PropTypes.string,
};

export default RunningTitle;
