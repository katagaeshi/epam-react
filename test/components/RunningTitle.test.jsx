import React from 'react';
import renderer from 'react-test-renderer';
import RunningTitle from './../../public/components/RunningTitle';

describe('RunningTitle', () => {
  it('Snapshot', () => {
    const tree = renderer
      .create(<RunningTitle />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
