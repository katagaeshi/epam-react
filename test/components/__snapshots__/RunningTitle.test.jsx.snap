// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunningTitle Snapshot 1`] = `
<span
  className="RunningTitle"
>
  netflixroulette
</span>
`;
